### PR TITLE
Utreexo docs

### LINKED ISSUE #275: [dev] Document specific version of utreexo we are implementing

### PR DESCRIPTION
**Pull request by m-kus**
- [x] resolves #275 
- [ ] follows contribution [guide](https://github.com/keep-starknet-strange/raito/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests

### CODE REVIEW

File: docs/img/utreexo_pollard.svg
Comment (maciejka) [inline docs/img/utreexo_pollard.svg:1]: Node `e` seems to be duplicated. I would also add an explanation like: `a` removal, `e` addition

File: docs/utreexo.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+Stump is the "light" version of Utreexo accumulator that contains the minimal necessary data required to verify inclusion proofs and add/remove leaves. It is not sufficient to generate proofs though — for that you need the full pollard, which is quite large.  
+Stump is a perfect fit for a ZK client where maintaining large state is prohibitively expensive.
+
+More specifically, stump consists of roots and the number of leaves (as we know this is the total number of UTXOs). Note that there are differences in how roots are stored and represented compared to the vanilla implementation:
+- Roots are sorted top down
+- There are no placeholders for rows that do not have roots
+- Empty roots are legit, it just means its children (aka nieces) are empty too
+
+<p align="center" width="100%">
+  <img src="./img/utreexo_stump.svg" alt=""/>
```
Comment (maciejka) [inline docs/utreexo.md:49]: What dotted border represents?

File: docs/utreexo.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+Note that previously we had either a root or "null" for every row in the forest, but in stump we have to use a different approach to determine to which row a particular root belongs. A simple rule is if the row length is even we skip it, otherwise (odd) the root is added. Another way to see this is to look at the binary representation of the leaves number and for each "1" there should be a root in the according row.
+
+Also it's worth noting that before "null" was indicating a missing roots whereas in stump empty root just means its children are empty too.
+
+## Batch proof
+
+Batch inclusion proofs were introduced in the Utreexo paper (see Appendix) and they can reduce the total size of proof(s) significantly. But the batch delete algorithm was pretty complex and IO intensive. The pollard scheme streamlined batch deletions and made batch proofs more practical.
+
+<p align="center" width="100%">
+  <img src="./img/utreexo_proof.svg" alt=""/>
```
Comment (maciejka) [inline docs/utreexo.md:61]: It is not clear what is being proven here.

File: docs/img/utreexo_pollard.svg
Comment (m-kus) [reply docs/img/utreexo_pollard.svg:1]: Fixed, updated the image

File: docs/utreexo.md (hunk 2 above)
Comment (m-kus) [reply docs/utreexo.md:64]: Inclusion :) I've added a sentence what batch proofs are consist of

**Review by maciejka — APPROVED** [review]

**Pull request merged**